app/src/main/java/com/rai/hearthstonecard/adapter/CardAdapter.kt

```diff
@@ -9,20 +9,23 @@ import androidx.recyclerview.widget.RecyclerView
 import coil.load
 import com.rai.hearthstonecard.databinding.ItemCardBinding
 import com.rai.hearthstonecard.databinding.ItemLoadingBinding
-import com.rai.hearthstonecard.retrofit.CardItem
+import com.rai.hearthstonecard.retrofit.Card
+import com.rai.hearthstonecard.retrofit.Item
 
 
 class CardAdapter(
     context: Context,
-    private val onItemClicked: (CardItem.Card) -> Unit,
-) : ListAdapter<CardItem, RecyclerView.ViewHolder>(DIFF_UTIL) {
+    private val onItemClicked: (Item.Content<Card>) -> Unit,
+) : ListAdapter<Item<*>, RecyclerView.ViewHolder>(DIFF_UTIL) {
```

Comment:
использование wildcards в классах такая себе затея, лучше параметризовать 
```suggestion
) : ListAdapter<Item<Card>, RecyclerView.ViewHolder>(DIFF_UTIL) {
```
либо если нужна более общая структура, то параметризовать адаптер `ItemAdapter<T>`